exonize/exonize_handler.py

```diff
@@ -1240,34 +1337,105 @@ def run_exonize_pipeline(self) -> None:
-            for indx in range(0, it_length, n):
-                yield iterable[indx:min(indx + n, it_length)]
-
-        self.prepare_data()
-        args_list = list(self.gene_hierarchy_dict.keys())
-        processed_gene_ids = query_gene_ids_in_res_db(self.results_db, self.timeout_db)
-        if processed_gene_ids:
-            args_list = [i for i in args_list if i not in processed_gene_ids]
-        if args_list:
-            batches_list = [i for i in batch(args_list, self.batch_number)]
+            # We round up to the upper integer value to guarantee that there will be 'number_of_batches' batches
+            even_batch_size = (len(data) // number_of_batches) + 1
+            for batch_number in range(number_of_batches):
+                batch_start_index = batch_number * even_batch_size
+                batch_end_index = min((batch_number + 1) * even_batch_size, len(data))
+                yield data[batch_start_index:batch_end_index]
+
+        self.prepare_data(pickled_filepath=self.genome_pickled_filepath)
+        gene_ids = list(self.gene_hierarchy_dict.keys())
+        processed_gene_ids = set(query_gene_ids_in_res_db(self.results_db, self.timeout_db))
+        unprocessed_gene_ids = [i for i in gene_ids if i not in processed_gene_ids]
+        if unprocessed_gene_ids:
+            self.logger.info(f'- Starting tblastx search for {len(unprocessed_gene_ids)} unprocessed genes.')
             gene_n = len(list(self.gene_hierarchy_dict.keys()))
-            self.logger.info(f'Starting exon duplication search for {len(args_list)}/{gene_n} genes.')
-            with tqdm(total=len(args_list), position=0, leave=True, ncols=50) as progress_bar:
-                for arg_batch in batches_list:
-                    t = ThreadPool(processes=self.threads)
-                    t.map(self.find_coding_exon_duplicates, arg_batch)
-                    t.close()
-                    t.join()
-                    progress_bar.update(len(arg_batch))
+            self.logger.info(f'- Starting exon duplication search for {len(unprocessed_gene_ids)}/{gene_n} genes.')
+            with tqdm(total=len(unprocessed_gene_ids), position=0, leave=True, ncols=50) as progress_bar:
+                # Benchmark without any parallel computation:
+                # pr = cProfile.Profile()
+                # pr.enable()
+                # for gene_id in unprocessed_gene_ids:
+                #     self.find_coding_exon_duplicates(gene_id)
+                #     progress_bar.update(1)
+                # pr.disable()
+                # pr.dump_stats(PROFILE_PATH)
+                # get_run_performance_profile(PROFILE_PATH)
+
+                # Benchmark with parallel computation using os.fork:
+                pr = cProfile.Profile()
+                pr.enable()
+                gc.collect()
+                gc.freeze()
+                transactions_pks: set[int]
+                status: int
+                code: int
+                forks: int = 0
```

Comment:
Is this necessary, i.e. including the variable types? I have not done this throughout my code, should I?